### PR TITLE
Upgrade Python / linter versions for development

### PR DESCRIPTION
**Pull request by bonprosoft**
Sorry for such a big PR :bowing_man: 

I upgraded the linter versions for the pysen development to the latest ones.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,10 +10,22 @@ enable_flake8 = true
...
+    "tests/",
+  ]
+[[tool.pysen.lint.mypy_targets]]
+  paths = ["examples/advanced_example"]
+[[tool.pysen.lint.mypy_targets]]
+  paths = ["examples/simple_package"]
+[[tool.pysen.lint.mypy_targets]]
+  paths = ["examples/plugin_example"]
+[[tool.pysen.lint.mypy_targets]]
+  paths = ["examples/sync_cmdclass_pyproject"]
```
Comment (bonprosoft) [inline pyproject.toml:28]: As mypy 0.800+ changes the way to find target sources, I updated this section as described in our FAQ https://github.com/pfnet/pysen#frequently-asked-questions .

**Review by sergeant-wizard — APPROVED** [review]
lgtm

**Comment by bonprosoft** [comment]
@sergeant-wizard Thank you for your review! I reverted some of the changes to keep backward compatibility. PTAL :bowing_man: 
https://github.com/pfnet/pysen/pull/26/commits/4c2c2db3d952dcc6dbf6ffe5708a2651fd9b608b

**Review by sergeant-wizard — APPROVED** [review]
lgtm

**Pull request merged**